pkg/helpers/cmd/errors.go

```diff
@@ -30,3 +35,80 @@ func CheckPodSecurityErr(err error) {
 
 	kcmdutil.CheckErr(err)
 }
+
+func CustomOIDCDerivedErrorChecker(discoveryClient discovery.DiscoveryInterface, what string) func(error) {
+	return func(err error) {
+		CheckGroupResourceNotRegisteredErr(err, discoveryClient, what, CustomOIDCReason, oauthv1.GroupVersion.Group, userv1.GroupVersion.Group)
+	}
+}
+
+const (
+	CustomOIDCReason = "This may occur if an external OIDC provider is configured."
+)
+
+// CheckGroupResourceNotRegisteredErr checks to see if originalError stems from the fact that
+// the GR in question is not registered on the server.
+func CheckGroupResourceNotRegisteredErr(originalError error, discoveryClient discovery.DiscoveryInterface, what, why string, groups ...string) {
+	if agg, ok := originalError.(utilerrors.Aggregate); ok && len(agg.Errors()) == 1 {
+		originalError = agg.Errors()[0]
+	}
+
+	if originalError == nil {
+		return
+	}
+
+	// only NotFound errors can stem from missing GRs, so short-circuit if we're
+	// dealing with any other error
+	if !apierrors.IsNotFound(originalError) {
+		kcmdutil.CheckErr(originalError)
+		return
+	}
+
+	status, ok := originalError.(apierrors.APIStatus)
+	if !ok && !errors.As(originalError, &status) {
+		// this should never happen, since `apierrors.IsNotFound()` succeeded above
+		kcmdutil.CheckErr(originalError)
+		return
+	}
+	details := status.Status().Details
+	if details == nil {
+		// we're not going to be able to detect whether the GR is registered
+		kcmdutil.CheckErr(originalError)
+		return
+	}
+
+	if len(groups) > 0 && !sets.New[string](groups...).Has(details.Group) {
+		// not a group we care about
+		kcmdutil.CheckErr(originalError)
+		return
+	}
+
+	// in the case of a NotFound error, either the GR is not found or the object iself
+	// is not found (when the action was a mutation of existing content); make sure it's
+	// a missing GR by cross-checking discovery
+	_, resourceLists, err := discoveryClient.ServerGroupsAndResources()
+	// ServerGroupsAndResources() can return partial data alongside an error, so first check
+	// for the GR - if we find it, we know that the culprit is not a missing resource
+	for _, resourceList := range resourceLists {
+		for _, resource := range resourceList.APIResources {
+			// n.b. it seems like metav1.Status.Details.Kind is a resource, not a kind
+			// n.b. metav1.Status does not specify a version, so the best we can do is detect that
+			// *some* related GroupResource (at some version) exists on the server
+			if resource.Name == details.Kind && resource.Group == details.Group {
+				// the GVK does exist on the server, so the NotFound must have been for a missing resource
+				kcmdutil.CheckErr(originalError)
+				return
+			}
+		}
+	}
+	if err != nil {
```

Comment:
What we lose if we ignore this error check?. Because it may change the error message from user not found to more non graceful.

Reply:
If this error occurred, we can't be sure that not finding the GR in discovery really means the resource is missing, so we fall back to `kcmdutil.CheckErr(originalError)` - which is the original error, not the one from discovery, so nothing is changed, I think?

Reply:
You are right, I missed `kcmdutil.CheckErr(originalError)`